samples/subsys/tracing/src/gpio_main.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_emul.h>
+
+
+/* size of stack area used by each thread */
+#define STACKSIZE (2048)
+
+/* scheduling priority used by each thread */
+#define PRIORITY 7
+
+
+void test_handler(const struct device *port, struct gpio_callback *cb, uint32_t pins)
+{
+	printk("Interrupt detected!\n");
+}
+
+
+void gpio_sample(void)
+{
+	const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(gpio0));
```

Comment:
You can create a unique gpio-emul node in `app.overlay` so that this can work in every boards and don't have to rely on native-sim